cli/commands/create/create_test.go

```diff
@@ -0,0 +1,78 @@
+package create
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"text/template"
+
+	"github.com/sensu/sensu-go/types"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	mockclient "github.com/sensu/sensu-go/cli/client/testing"
+	cmdtesting "github.com/sensu/sensu-go/cli/commands/testing"
+)
+
+var resourceSpecTmpl = template.Must(template.New("test").Parse(`
+{"type": "Check", "spec": {{ .Check }} }
+{"type": "Asset", "spec": {{ .Asset }} }
+{"type": "Hook", "spec": {{ .Hook }} }
+`))
+
+func mustMarshal(t interface{}) string {
+	b, err := json.Marshal(t)
+	if err != nil {
+		panic(err)
+	}
+	return string(b)
+}
+
+var (
+	fixtureCheck = types.FixtureCheck("foo")
+	fixtureAsset = types.FixtureAsset("bar")
+	fixtureHook  = types.FixtureHook("baz")
+)
+
+var resources = struct {
+	Check string
+	Asset string
+	Hook  string
+}{
+	Check: mustMarshal(fixtureCheck),
+	Asset: mustMarshal(fixtureAsset),
+	Hook:  mustMarshal(fixtureHook),
+}
+
+func TestCreateCommand(t *testing.T) {
+	cli := cmdtesting.NewMockCLI()
+	client := cli.Client.(*mockclient.MockClient)
+	client.On("PutResource", mock.AnythingOfType("*types.Check")).Return(nil)
```

Comment:
Is it possible to validate that we're correctly deserializing and not just getting zero values here?

Reply:
I can add a test case where validation would fail.